### ISSUE TITLE
Closing a tab exits fullscreen videos

### ISSUE THREAD

**Issue by ssorallen** [tabwrangler/tabwrangler#236]
This was reported in [review 1140451](https://addons.mozilla.org/en-US/firefox/addon/tabwrangler/reviews/1140451/) on the Firefox AMO:

> When watching fullscreen youtube videos, when the extension decides to close a tab, it takes you out of fullscreen, and this can be incredibly annoying especially when a bunch of tabs close all at once.

**Comment by ingorichter** [comment]
yeah, this is really annoying. I wonder if postponing the close operation until Firefox is no longer in full screen mode could mitigate this behavior. I would consider this behavior a bug in Firefox. On the other hand who closes tabs when in fullscreen mode. This sounds more like an unusual use case and workflow to me. So my first thought about postponing might be a way to deal with it

**Comment by ssorallen** [comment]
@ingorichter I agree this seems like unexpected behavior in FF. I haven’t yet checked whether there’s a bug/feature request for it. In the meantime, your suggestion seems like a great workaround. We could use window.fullscreen:

https://developer.mozilla.org/en-US/docs/Web/API/Window/fullScreen

**Comment by jasikpark** [comment]
On macOS it's possible to close browser tabs while in fullscreen due to the fact that the fullscreen tab becomes another workspace. So I can then switch to the original normal Firefox window and close whatever tabs I want. I suppose since I'm already looking at the window, the command to bring the window into focus is not visible to me. 

What probably would need to happen is for Firefox to update with a guard for whether closing tabs draws you back to the window based on whether the tab that is closed is currently fullscreen

**Comment by jessecrouch** [comment]
Would love to see an option for "no close when in full screen" implemented. Gonna try "don't close tabs playing audio" for now.
